glib-macros/src/gboxed_derive.rs

```diff
@@ -116,28 +118,11 @@ pub fn impl_gboxed(input: &syn::DeriveInput) -> TokenStream {
     quote! {
         impl #crate_ident::subclass::boxed::BoxedType for #name {
             const NAME: &'static str = #gtype_name;
-
-            fn type_() -> #crate_ident::Type {
-                static mut TYPE_: #crate_ident::Type = #crate_ident::Type::INVALID;
-                static ONCE: ::std::sync::Once = ::std::sync::Once::new();
-
-                ONCE.call_once(|| {
-                    let type_ = #crate_ident::subclass::register_boxed_type::<Self>();
-                    unsafe {
-                        TYPE_ = type_;
-                    }
-                });
-
-                unsafe {
-                    assert!(TYPE_.is_valid());
-                    TYPE_
-                }
-            }
         }
 
         impl #crate_ident::StaticType for #name {
             fn static_type() -> #crate_ident::Type {
-                <#name as #crate_ident::subclass::boxed::BoxedType>::type_()
+                #get_type()
```

Comment:
Any reason why we have to call `#get_type()` here instead of just having the function content here? is it useful for something else?

Reply:
That's how it's done in all the other derive macros... but good point. Let's just inline it here

Reply:
(and in the other macros)

Reply:
I blatantly copied from GEnum :)